eva-server/src/test/java/uk/ac/ebi/eva/server/ws/GeneWSServerV2Test.java

```diff
@@ -0,0 +1,164 @@
+                .willReturn(Arrays.asList(variantEntity, variantEntity));
+        given(variantService.countByRegionsAndComplexFilters(eq(twoRegions), any())).willReturn(2l);
+
+        given(variantService
+                .findByRegionsAndComplexFilters(not(or(eq(oneRegion), eq(twoRegions))), any(), any(), any(), any()))
+                .willReturn(Collections.emptyList());
+        given(variantService.countByRegionsAndComplexFilters(not(or(eq(oneRegion), eq(twoRegions))), any()))
+                .willReturn(0l);
+
+        FeatureCoordinates feature1 = new FeatureCoordinates(GENE_ID1, "id", "feature", "20", 60000L, 62000L);
+        FeatureCoordinates feature2 = new FeatureCoordinates(GENE_ID1, "id", "feature", "20", 63000L, 64000L);
+
+        List<FeatureCoordinates> featureCoordinate = Arrays.asList(feature1);
+        List<String> geneId = Arrays.asList(GENE_ID1);
+        given(featureService.findAllByGeneIdsOrGeneNames(eq(geneId), eq(geneId))).willReturn(featureCoordinate);
+
+        List<FeatureCoordinates> featureCoordinates = Arrays.asList(feature1, feature2);
+        List<String> geneIds = Arrays.asList(GENE_ID1, GENE_ID2);
+        given(featureService.findAllByGeneIdsOrGeneNames(eq(geneIds), eq(geneIds))).willReturn(featureCoordinates);
+    }
+
+    @Test
+    public void testGetVariantsByExistingGene() throws URISyntaxException {
+        assertEquals("20", testGetVariantsGeneHelper("ENSG00000227232", 1, HttpStatus.OK).get(0));
+    }
+
+    private List<String> testGetVariantsGeneHelper(String testRegion, int expectedVariants, HttpStatus status)
+            throws URISyntaxException {
+        String url = "/v2/genes/" + testRegion + "/variants?species=mmusculus&assembly=grcm38";
+        ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+        assertEquals(status, response.getStatusCode());
+        if (status == HttpStatus.NO_CONTENT) {
+            assertNull(response.getBody());
+            return null;
+        }
+        Configuration configuration = Configuration.defaultConfiguration()
+                .jsonProvider(new JacksonJsonProvider())
+                .mappingProvider(new JacksonMappingProvider(objectMapper))
+                .addOptions(Option.SUPPRESS_EXCEPTIONS);
+        List<Variant> variantList = JsonPath.using(configuration).parse(response.getBody())
+                .read("$['_embedded']['variantList']", new TypeRef<List<Variant>>() {
+                });
+
+        if (variantList == null) {
+            assertEquals(0, expectedVariants);
+            return null;
+        }
+        assertEquals(expectedVariants, variantList.size());
+        List<String> chromosomes = new ArrayList<>();
+        variantList.forEach(variant -> {
+            chromosomes.add(variant.getChromosome());
+        });
+        return chromosomes;
+    }
+
+    @Test
+    public void testGetVariantsByExistingGenes() throws URISyntaxException {
```

Comment:
Even though is ok to check the chromosomes the variants should be checked as well

Reply:
Added more checks :+1: